source/common/common/BUILD

```diff
@@ -179,8 +179,8 @@ envoy_cc_library(
 )
 
 envoy_cc_library(
-    name = "shared_memory_hash_set_lib",
```

Comment:
@mrice32 This is one of a few commits which rename `SharedMemoryHashSet` to `MemoryHashSet`, or `shared_memory_hash_set` to `memory_hash_set`, or `sharedMemOptions` to `memOptions`. I wanted to indicate that the MemoryHashSet doesn't need to be run on shared memory.